app/code/Magento/Catalog/Model/Indexer/Product/Flat/Action/Row.php

```diff
@@ -85,10 +85,16 @@ public function execute($id = null)
         $ids = [$id];
         $linkField = $this->metadataPool->getMetadata(ProductInterface::class)->getLinkField();
 
+        $storeIds = $this->getProductAvailableStores($id);
```

Comment:
I think the function should be changed to name `getAvailableStoresByProductId`. So it means we get Store Ids list instead of get product.

Reply:
Thanks for your recommendation, I decided to change function's name to getAssignedStoreIdsOfProduct